### PR TITLE
Created Horovod.run

### PR DESCRIPTION
**Pull request by abditag2**
Now, we can use the following to launch a job. 
```python -m horovod -np 3 -H worker-0:1,worker-1:1,worker-2:1 model.py```

### CODE REVIEW

**Comment by alsrgv** [comment]
Thanks, Fardin!  First UX questions for starters:

1. Can you refactor the command to be `horovodrun -np 3 -H ... python model.py`?  [This doc](https://python-packaging.readthedocs.io/en/latest/command-line-scripts.html) should help with implementation ideas.

2. What's the performance benefit of caching?  How long does cache last?  Server may reconfigure interfaces from time to time.

**Comment by tgaddair** [comment]
> Thanks, Fardin! First UX questions for starters:
> 
> 1. Can you refactor the command to be `horovodrun -np 3 -H ... python model.py`?  [This doc](https://python-packaging.readthedocs.io/en/latest/command-line-scripts.html) should help with implementation ideas.
> 2. What's the performance benefit of caching?  How long does cache last?  Server may reconfigure interfaces from time to time.

Thoughts on name for the executable?  I think `hvdrun` would be preferable to `horovodrun` (shorter, and that's already the abbreviation we suggest to users, similar to `k8s` for kubernetes).

**Comment by alsrgv** [comment]
I'm probably biased, but I like `horovodrun` :-)  For k8s, you actually use `kubectl`.

File: horovod/__init__.py [hunk 1]
```diff
@@ -1 +1,236 @@
 __version__ = '0.16.0'
+
```
Comment (alsrgv) [inline horovod/__init__.py]: Move out

File: horovod/__init__.py [hunk 2]
```diff
@@ -1 +1,236 @@
...
+    parser.add_argument('-H', '--host', action="store", dest="host",
+                        help="To specify the list of hosts on which "
+                             "to invoke processes. Takes a comma-delimited "
+                             "list of hosts.")
+
+    parser.add_argument('--disable-cache', action="store_true",
+                        dest="disable_cache",
+                        help="If flag is set, horovod run will cache the "
+                             "resutls and use them.")
```
Comment (tgaddair) [inline horovod/__init__.py]: s/resutls/results

File: horovod/run/util.py [hunk 3]
```diff
@@ -0,0 +1,213 @@
...
+            params['hostname'] = options['hostname']
+            params['look_for_keys'] = False
+            if 'port' in options.keys():
+                params['port'] = options['port']
+            self.client.connect(**params)
+
+        self.transport = paramiko.Transport((options['hostname'], 22))
+        self.transport.connect()
+
+    def closeConnection(self):
```
Comment (alsrgv) [inline horovod/run/util.py]: Switch to pythonic names

File: horovod/run/util.py [hunk 4]
```diff
@@ -0,0 +1,213 @@
+import paramiko
+import time
+import socket
+import os
+import Queue
```
Comment (alsrgv) [inline horovod/run/util.py]: Py2/Py3 use six

File: horovod/__init__.py [hunk 5]
```diff
@@ -1 +1,236 @@
...
+                client_module=CLIENT_MODULE, python=sys.executable,
+                hvd_secred=hvd_secret_bs64,
+                server_name_bs64=target_address_bs64,
+                addresses_bs64=codec.dumps_base64(intfc_addr_to_probe))
+
+            ssh_host.sendShell(command)
+            response = ssh_host.recv()
+
+            addresses = util.check_success_msg_and_extract_result(response,
+                                                                  'CLIENT LAUNCH SUCCESSFUL',
```
Comment (alsrgv) [inline horovod/__init__.py]: Super hacky, we should switch to more reliable communication

File: horovod/run/util.py [hunk 6]
```diff
@@ -0,0 +1,213 @@
...
+    transport = None
+
+    def __init__(self, hostname):
+
+        self.client = paramiko.client.SSHClient()
+        self.client.set_missing_host_key_policy(paramiko.client.AutoAddPolicy())
+        try:
+            self.client.connect(hostname, look_for_keys=False)
+        except socket.gaierror:
+            config = paramiko.SSHConfig()
```
Comment (alsrgv) [inline horovod/run/util.py]: We probably want to switch back to using stock `ssh` to align how configuration (e.g. hostname mapping) is digested.

**Comment by alsrgv** [comment]
Superseded by #869

**Pull request closed without merging**